newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -0,0 +1,565 @@
+
+const electron = optionalRequire('electron');
+const path = optionalRequire('path');
+const app = electron ? electron.remote.app : null;
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    width: '100%',
+  },
+  content: {
+    display: 'flex',
+    flexDirection: 'column',
+    flex: 1,
+    maxWidth: 1200,
+    marginTop: 10,
+    marginBottom: 20,
+    padding: 0,
+    alignSelf: 'center',
+    width: '100%',
+  },
+};
+
+type Props = {|
+  project: ?gdProject,
+
+  isActive: boolean,
+  projectItemName: ?string,
+  project: ?gdProject,
+  setToolbar: (?React.Node) => void,
+
+  // Project opening
+  canOpen: boolean,
+  onOpen: () => void,
+  onOpenRecentFile: (file: FileMetadataAndStorageProviderName) => void,
+  onOpenExamples: () => void,
+  onOpenProjectManager: () => void,
+  onCloseProject: () => Promise<void>,
+
+  // Other dialogs opening:
+  onOpenTutorials: () => void,
+  onOpenGamesShowcase: () => void,
+  onOpenHelpFinder: () => void,
+  onOpenLanguageDialog: () => void,
+  onOpenProfile: () => void,
+
+  // Project creation
+  onCreateFromExampleShortHeader: OnCreateFromExampleShortHeaderFunction,
+  onCreateBlank: OnCreateBlankFunction,
+  onOpenProjectAfterCreation: OnOpenProjectAfterCreationFunction,
+|};
+
+type HomePageEditorInterface = {|
+  getProject: () => void,
+  updateToolbar: () => void,
+  forceUpdateEditor: () => void,
+|};
+
+const betweenCarouselSpacerCount = 6;
+
+const renderBetweenCarouselSpace = (offset: number = 0) =>
+  Array(betweenCarouselSpacerCount)
+    .fill(0)
+    .map((e, index) => <Spacer key={`spacer${index + offset}`} />);
+
+export const HomePage = React.memo<Props>(
+  React.forwardRef<Props, HomePageEditorInterface>(
+    (
+      {
+        project,
+        canOpen,
+        onOpen,
+        onOpenRecentFile,
+        onCreateFromExampleShortHeader,
+        onCreateBlank,
+        onOpenProjectAfterCreation,
+        onOpenExamples,
+        onOpenProjectManager,
+        onCloseProject,
+        onOpenTutorials,
+        onOpenGamesShowcase,
+        onOpenHelpFinder,
+        onOpenLanguageDialog,
+        onOpenProfile,
+        setToolbar,
+      }: Props,
+      ref
+    ) => {
+      const getProject = () => {
+        return undefined;
+      };
+
+      const updateToolbar = () => {
+        if (setToolbar) setToolbar(null);
+      };
+
+      const forceUpdateEditor = () => {
+        // No updates to be done
+      };
+
+      React.useImperativeHandle(ref, () => ({
+        getProject,
+        updateToolbar,
+        forceUpdateEditor,
+      }));
+
+      const windowWidth = useResponsiveWindowWidth();
+      const authenticatedUser = React.useContext(AuthenticatedUserContext);
+      const { getRecentProjectFiles } = React.useContext(PreferencesContext);
+      const [
+        recentProjectFiles,
+        setRecentProjectFiles,
+      ] = React.useState<?Array<FileMetadataAndStorageProviderName>>(null);
```

Comment:
@AlexandreSi I think you might not need a state for this, `getRecentProjectFiles` is safe to read at any time and should always return an array, even if empty.